xds/internal/xdsclient/creds/tls.go

```diff
@@ -0,0 +1,177 @@
+/*
+ *
+ * Copyright 2023 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package creds implements gRFC A65: mTLS Credentials in xDS Bootstrap File.
+package creds
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net"
+	"sync"
+
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/tls/certprovider"
+	_ "google.golang.org/grpc/credentials/tls/certprovider/pemfile" // for file_watcher provider
+)
+
+type tlsBundle struct {
+	jd                   json.RawMessage
+	transportCredentials credentials.TransportCredentials
+}
+
+// NewTLS returns a credentials.Bundle which delegates certificate loading to
+// a file_watcher provider for mTLS transport security. See gRFC A65.
```

Comment:
```suggestion
// a file_watcher provider for mTLS transport security.
// See [gRFC A65](github.com/grpc/proposal/blob/master/A65-xds-mtls-creds-in-bootstrap.md).
```

Reply:
I rephrased this comment, hopefully keeping the spirit of your comment.

Reply:
Thanks.